C/include/c4Database.h

```diff
@@ -171,12 +173,12 @@ CBL_CORE_API bool c4db_maintenance(C4Database* database, C4MaintenanceType type,
 
 /** Begins a transaction.
         Transactions can nest; only the first call actually creates a database transaction. */
-CBL_CORE_API bool c4db_beginTransaction(C4Database* database, C4Error* C4NULLABLE outError) C4API;
+NODISCARD CBL_CORE_API bool c4db_beginTransaction(C4Database* database, C4Error* C4NULLABLE outError) C4API;
 
 /** Commits or aborts a transaction. If there have been multiple calls to beginTransaction, it
         takes the same number of calls to endTransaction to actually end the transaction; only the
         last one commits or aborts the database transaction. */
-CBL_CORE_API bool c4db_endTransaction(C4Database* database, bool commit, C4Error* C4NULLABLE outError) C4API;
+NODISCARD CBL_CORE_API bool c4db_endTransaction(C4Database* database, bool commit, C4Error* C4NULLABLE outError) C4API;
```

Comment:
CBL Swift complains about this NODISCARD on db_endTransaction. If removed, it builds just fine on both iOS and macOS

Reply:
That's strange, because it's syntactically just like many of the other changes, `NODISCARD CBL_CORE_API <type> ...`.
What is the exact error?

Reply:
fixed on platform by just casting void to the destructor as discussed